app/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_experiment_branch_mixin.py

```diff
@@ -264,35 +264,73 @@ def test_no_duplicate_branch_names(self):
         )
 
     def test_no_treatment_branches_for_rollout(self):
-        experiment = NimbusExperimentFactory.create(
-            status=NimbusExperiment.Status.DRAFT,
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            application=NimbusExperiment.Application.DESKTOP,
+            channel=NimbusExperiment.Channel.NO_CHANNEL,
+            feature_configs=[
+                NimbusFeatureConfigFactory(
+                    application=NimbusExperiment.Application.DESKTOP
+                )
+            ],
+            is_sticky=True,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.MAC_ONLY,
         )
         experiment.is_rollout = True
         experiment.save()
         for branch in experiment.treatment_branches:
             branch.delete()
 
         data = {
+            "application": f"{NimbusExperiment.Application.DESKTOP}",
+            "is_sticky": "false",
+            "targeting_config_slug": f"{NimbusExperiment.TargetingConfig.MAC_ONLY}",
             "treatment_branches": [
                 {"name": "treatment A", "description": "desc1", "ratio": 1},
                 {"name": "treatment B", "description": "desc2", "ratio": 1},
             ],
             "changelog_message": "test changelog message",
+            "channel": "",
         }
         serializer = NimbusExperimentSerializer(
             experiment, data=data, partial=True, context={"user": self.user}
         )
+
         self.assertFalse(serializer.is_valid())
-        self.assertEqual(
-            serializer.errors,
-            {
-                "treatment_branches": [
-                    {"name": NimbusConstants.ERROR_SINGLE_BRANCH_FOR_ROLLOUT}
-                    for i in data["treatment_branches"]
-                ],
-            },
+        self.assertEqual(len(serializer.errors), 1)
+
+        error = serializer.errors["treatment_branches"][0].get("name")
```

Comment:
See just reiterating what I was saying earlier, here `"name"` may or may not be in the dict, so we either get back the error or None.  We check it on the next line so we'll still see a test failure so in theory either works here, but it's also okay to just assume it's there for the test and let it throw as the test failure.